### PR TITLE
/github signout

### PR DESCRIPTION
**Pull request by wilhelmklopp**
Add `/github signout` as an undocumented command to help with discovering edge cases and as an option in case users accidentally? link the wrong GitHub account

Todo:
- [x] Get pending command to not loop infinitely
- [x] add tests
- [x] Figure out other implications of sign out. Do we delete subscriptions automatically?
- [x] delete subscriptions
- [ ]~post interactive message with 'confirm' option. Also have an option that includes: 'Post message in each channel prompting to re-enable all disabled subscriptions'~ not doing this right now because of the effort needed to make it happen
- [x] make signing out a method on the model so that it can be re-used when initiated by `github_app_authorization.revoked` in https://github.com/integrations/slack/issues/446

### CODE REVIEW

File: lib/commands/signout.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+const { SignOut } = require('../messages/flow');
+
+const SignedParams = require('../signed-params');
+
+module.exports = async (req, res) => {
+  await res.locals.slackUser.update({
```
Comment (bkeepers) [inline lib/commands/signout.js]: Nit: since there are a few `res.locals` here, wdyt about pulling them out as variables first so it's obvious what dependencies this has?

File: lib/commands/signout.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  });
+
+  const { command } = res.locals;
+
+  const state = new SignedParams({
+    trigger_id: command.trigger_id,
+  });
+
+  const protocol = req.headers['x-forwarded-proto'] || req.protocol;
+  const host = req.headers['x-forwarded-host'] || req.get('host');
```
Comment (bkeepers) [inline lib/commands/signout.js]: What do you think about pulling this out into a module? I think we do this in 2-3 places now.

File: lib/messages/flow/signout.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  toJSON() {
+    return {
+      response_type: 'ephemeral',
+      attachments: [{
+        ...this.getBaseMessage(),
+        fallback: 'Your GitHub account has been disconnected.',
+        text: `:white_check_mark: <@${this.userSlackId}> is disconnected from your GitHub account\n` +
+          'Features like subscriptions and unfurls will stop working. ' +
+          'To re-connect your GitHub account click the button below.',
```
Comment (bkeepers) [inline lib/messages/flow/signout.js]: I know we're already using  "connect" elsewhere, but what about switching to the language of "signed in" and "signed out". I also don't think we've been using "unfurls" in any user-facing language.

> :white_check_mark: <@U98765> is now signed out. Use `/github signin` to sign back in to your GitHub account at any time.

🤷‍♂️ 

@cmwinters, what do you think?

Comment (cmwinters) [reply lib/messages/flow/signout.js]: @bkeepers seems fine to me.

Comment (wilhelmklopp) [reply lib/messages/flow/signout.js]: 👍 Also prefer to avoid "connect". I think it is valuable to tell the user that features will stop working though. That might not be obvious to them.

File: lib/commands/signout.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      const subscriptions = subscriptionsByChannel[channel];
+      const repositories = await getRepositories(subscriptions, robot);
+      if (repositories) {
+        const attachment = new ReEnableMultipleSubscriptions(repositories, slackUser.slackId, 'disconnected their GitHub account').getAttachment();
+        return slackWorkspace.client.chat.postMessage({
+          channel,
+          attachments: [attachment],
+        });
+      }
+    }));
```
[the remark this answers is not in the document]
Comment (wilhelmklopp) [reply lib/commands/signout.js]: Extra context: I plan to refactor this code, so that it will be used both by `/github signout` AND the code for the "token revoked" event. That event will probably lead to this code path being called much more frequently: https://github.com/integrations/slack/issues/446

In a way this PR was just a set up for this follow up one
